File: Java-Core/src/main/java/presto/javascript/JavaScriptItemExpression.java
package presto.javascript;

import presto.utils.CodeWriter;



public class JavaScriptItemExpression extends JavaScriptSelectorExpression {

	JavaScriptExpression item;
	
	public JavaScriptItemExpression(JavaScriptExpression item) {
		this.item = item;
	}

	@Override
	public String toString() {
		return parent.toString() + "[" + item.toString() + "]";
	}

	@Override
	public void toDialect(CodeWriter writer) {
		parent.toDialect(writer);
		writer.append('[');
		item.toDialect(writer);
		writer.append(']');
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/javascript/JavaScriptItemExpression.js
var JavaScriptSelectorExpression = require("./JavaScriptSelectorExpression").JavaScriptSelectorExpression;

function JavaScriptItemExpression(item) {
	JavaScriptSelectorExpression.call(this);
	this.item = item || null;
	return this;
}

JavaScriptItemExpression.prototype = Object.create(JavaScriptSelectorExpression.prototype);
JavaScriptItemExpression.prototype.constructor = JavaScriptItemExpression;

JavaScriptItemExpression.prototype.toString = function() {
	return this.parent.toString() + "[" + this.item.toString() + "]";
};

exports.JavaScriptItemExpression = JavaScriptItemExpression;